internal/controller/object/object.go

```diff
@@ -279,6 +350,27 @@ func (c *external) Delete(ctx context.Context, mg resource.Managed) error {
 		return err
 	}
 
+	if c.shouldWatch(cr) {
+		c.kindObserver.StopWatchingResources(ctx, cr.Spec.ProviderConfigReference.Name, obj.GroupVersionKind())
```

Comment:
isn't this the `Object`? surprises me to call the kind observer for it.

Reply:
This is the desired Kubernetes resource that is meant to be created not the `kind: Object`.